internal/config/config.go

```diff
@@ -82,22 +83,32 @@ func New(filename string, stores ...cfg.SecretStore) *Config {
 
 // Config represents main configuration.
 type Config struct {
-	ListenAddr string              `json:"listen"`             // The API port used for TCP & Websocket communication.
-	License    string              `json:"license"`            // The license file to use for the broker.
-	TLS        *cfg.TLSConfig      `json:"tls,omitempty"`      // The API port used for Secure TCP & Websocket communication.
-	Cluster    *ClusterConfig      `json:"cluster,omitempty"`  // The configuration for the clustering.
-	Storage    *cfg.ProviderConfig `json:"storage,omitempty"`  // The configuration for the storage provider.
-	Contract   *cfg.ProviderConfig `json:"contract,omitempty"` // The configuration for the contract provider.
-	Metering   *cfg.ProviderConfig `json:"metering,omitempty"` // The configuration for the usage storage for metering.
-	Logging    *cfg.ProviderConfig `json:"logging,omitempty"`  // The configuration for the logger.
-	Monitor    *cfg.ProviderConfig `json:"monitor,omitempty"`  // The configuration for the monitoring storage.
-	Vault      secretStoreConfig   `json:"vault,omitempty"`    // The configuration for the Hashicorp Vault Secret Store.
-	Dynamo     secretStoreConfig   `json:"dynamodb,omitempty"` // The configuration for the AWS DynamoDB Secret Store.
+	ListenAddr     string              `json:"listen"`             // The API port used for TCP & Websocket communication.
+	License        string              `json:"license"`            // The license file to use for the broker.
+	MaxMessageSize int               `json:"maxMessageSize"`     // Maximum message size allowed from/to the peer
+	TLS            *cfg.TLSConfig      `json:"tls,omitempty"`      // The API port used for Secure TCP & Websocket communication.
+	Cluster        *ClusterConfig      `json:"cluster,omitempty"`  // The configuration for the clustering.
+	Storage        *cfg.ProviderConfig `json:"storage,omitempty"`  // The configuration for the storage provider.
+	Contract       *cfg.ProviderConfig `json:"contract,omitempty"` // The configuration for the contract provider.
+	Metering       *cfg.ProviderConfig `json:"metering,omitempty"` // The configuration for the usage storage for metering.
+	Logging        *cfg.ProviderConfig `json:"logging,omitempty"`  // The configuration for the logger.
+	Monitor        *cfg.ProviderConfig `json:"monitor,omitempty"`  // The configuration for the monitoring storage.
+	Vault          secretStoreConfig   `json:"vault,omitempty"`    // The configuration for the Hashicorp Vault Secret Store.
+	Dynamo         secretStoreConfig   `json:"dynamodb,omitempty"` // The configuration for the AWS DynamoDB Secret Store.
 
 	listenAddr *net.TCPAddr     // The listen address, parsed.
 	certCaches []cfg.CertCacher // The certificate caches configured.
 }
 
+
+func (c *Config) MaxMessageBytes() int64{
```

Comment:
Instead of having `maxMessageSize` on the first level, we could create a `Limit` struct with `MessageSize` number inside. This would also read a bit better using the environment (i.e. `EMITTER_LIMIT_MESSAGESIZE`)

Reply:
agree - also we might be able to use it in future for other limits. Good idea would implement.

Reply:
fixed.